tests/d3d12_device.c

```diff
@@ -356,13 +559,75 @@ void test_format_support(void)
     ok(!format_support.Support2 || format_support.Support2 == D3D12_FORMAT_SUPPORT2_TILED,
             "Got unexpected support2 %#x.\n", format_support.Support2);
 
-    for (i = 0; i < ARRAY_SIZE(depth_stencil_formats); ++i)
+    memset(&feature_levels, 0, sizeof(feature_levels));
+    feature_levels.NumFeatureLevels = ARRAY_SIZE(all_feature_levels);
+    feature_levels.pFeatureLevelsRequested = all_feature_levels;
+    hr = ID3D12Device_CheckFeatureSupport(device, D3D12_FEATURE_FEATURE_LEVELS, 
+        &feature_levels, sizeof(feature_levels));
+    ok(hr == S_OK, "Got unexpected hr %#x.\n", hr);
+
+    for (k = 0; k < ARRAY_SIZE(dxgi_format_list); ++k)
```

Comment:
It's very awkward to use `k` here to propagate state into inner loops later. Preferably, you'd assign `&dxgi_format_list[k]` to some local pointer that's referenced later.